core/accessors.go

```diff
@@ -425,57 +410,71 @@ func GetBlockByNumber(r db.KeyValueReader, blockNum uint64) (*Block, error) {
 		return nil, err
 	}
 
-	txs, err := GetTxsByBlockNum(r, blockNum)
-	if err != nil {
-		return nil, err
-	}
-
-	receipts, err := GetReceiptsByBlockNum(r, blockNum)
+	blockTransactions, err := BlockTransactionsBucket.Get(r, blockNum)
 	if err != nil {
 		return nil, err
 	}
 
 	return &Block{
 		Header:       header,
-		Transactions: txs,
-		Receipts:     receipts,
+		Transactions: blockTransactions.Transactions,
+		Receipts:     blockTransactions.Receipts,
 	}, nil
 }
 
-func WriteTxAndReceipt(
+func WriteTransactionsAndReceipts(
 	w db.KeyValueWriter,
-	num, index uint64,
-	tx Transaction,
-	receipt *TransactionReceipt,
+	blockNumber uint64,
+	transactions []Transaction,
+	receipts []*TransactionReceipt,
 ) error {
-	txHash := (*felt.TransactionHash)(tx.Hash())
-	key := db.BlockNumIndexKey{
-		Number: num,
-		Index:  index,
+	for index, tx := range transactions {
+		txHash := (*felt.TransactionHash)(tx.Hash())
+		key := db.BlockNumIndexKey{
+			Number: blockNumber,
+			Index:  uint64(index),
+		}
+
+		if err := TransactionBlockNumbersAndIndicesByHashBucket.Put(w, txHash, &key); err != nil {
+			return err
+		}
 	}
 
-	if err := TransactionBlockNumbersAndIndicesByHashBucket.Put(w, txHash, &key); err != nil {
+	blockTransactions := BlockTransactions{
+		Transactions: transactions,
+		Receipts:     receipts,
+	}
+	if err := BlockTransactionsBucket.Put(w, blockNumber, &blockTransactions); err != nil {
 		return err
 	}
 
-	if err := TransactionsByBlockNumberAndIndexBucket.Put(w, key, &tx); err != nil {
-		return err
+	return nil
+}
+
+func GetTxByBlockNumberAndIndex(
```

Comment:
nitpick: some methods have been rewritten to use `Transaction` full name instead of `Tx`.

It's a very samll issue, but sometimes we use `Tx`, `Txn` or `Transaction`, as a way to standardize it to `Transaction`

Reply:
Updated